jerry-core/jerry.cpp

```diff
@@ -269,6 +269,48 @@ jerry_api_convert_api_value_to_ecma_value (ecma_value_t *out_value_p, /**< out:
   }
 } /* jerry_api_convert_api_value_to_ecma_value */
 
+/**
+ * Convert completion of 'eval' to API value and completion code
+ *
+ * Note:
+ *      if the output value contains string / object, it should be freed
+ *      with jerry_api_release_string / jerry_api_release_object,
+ *      just when it becomes unnecessary.
+ *
+ * @return completion code
+ */
+static jerry_completion_code_t
+jerry_api_convert_eval_completion_to_retval (jerry_api_value_t *retval_p, /**< out: api value */
+                                             ecma_completion_value_t completion) /**< completion of 'eval'-mode
+                                                                                  *   code execution */
+{
+  jerry_completion_code_t ret_code;
+
+  if (ecma_is_completion_value_normal (completion))
+  {
+    ret_code = JERRY_COMPLETION_CODE_OK;
+
+    jerry_api_convert_ecma_value_to_api_value (retval_p,
+                                               ecma_get_completion_value_value (completion));
+  }
+  else
+  {
+    jerry_api_convert_ecma_value_to_api_value (retval_p, ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED));
+
+    if (ecma_is_completion_value_throw (completion))
+    {
+      ret_code = JERRY_COMPLETION_CODE_UNHANDLED_EXCEPTION;
```

Comment:
It's reported to the caller as "unhandled", and the throw completion value is then freed outside of the conversion routine.